### PR TITLE
Initializing Clearance for authentication

### PR DESCRIPTION
**Pull request by orhanna14**

### CODE REVIEW

File: db/migrate/20191128003907_add_clearance_to_users.rb [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+    end
+
+    add_index :users, :remember_token
+
+    users = select_all("SELECT id FROM users WHERE remember_token IS NULL")
+
+    users.each do |user|
+      update <<-SQL
+        UPDATE users
+        SET remember_token = '#{Clearance::Token.new}'
```
Comment (whyderrick) [inline db/migrate/20191128003907_add_clearance_to_users.rb]: Nifty! Was this in the README or did you source dive to find this class name?

File: db/migrate/20191128003907_add_clearance_to_users.rb [hunk 2]
```diff
@@ -0,0 +1,26 @@
+class AddClearanceToUsers < ActiveRecord::Migration[6.0]
+  def self.up
+    change_table :users do |t|
+      t.string :confirmation_token, limit: 128
+      t.string :remember_token, limit: 128
```
Comment (whyderrick) [inline db/migrate/20191128003907_add_clearance_to_users.rb]: Should these be nullable? If not, add a constraint. If you allow them to become `null` they will eventually (in some hair-pulling way) end up as `null`.

File: db/migrate/20191128003907_add_clearance_to_users.rb [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+
+    add_index :users, :remember_token
+
+    users = select_all("SELECT id FROM users WHERE remember_token IS NULL")
+
+    users.each do |user|
+      update <<-SQL
+        UPDATE users
+        SET remember_token = '#{Clearance::Token.new}'
+        WHERE id = '#{user['id']}'
```
Comment (whyderrick) [inline db/migrate/20191128003907_add_clearance_to_users.rb]: There is a school of thought that avoids changing data in migrations. 

Here's a [good primer](https://thoughtbot.com/blog/data-migrations-in-rails). I don't think you should change this one, but I'd feel remiss if I didn't mention the other approach.

File: db/migrate/20191128003907_add_clearance_to_users.rb (hunk 1 above)
Comment (orhanna14) [reply db/migrate/20191128003907_add_clearance_to_users.rb]: This was in the README, created when I ran 'rails generate clearance:install'

File: db/migrate/20191128003907_add_clearance_to_users.rb (hunk 3 above)
Comment (orhanna14) [reply db/migrate/20191128003907_add_clearance_to_users.rb]: This is great, thanks! Again, this came from the generator, so I think I'll keep this for now, but bookmarked this for the future.

**Pull request merged**